src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/DefaultLogHubLogWriter.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Logging
+{
+    internal class DefaultLogHubLogWriter : LogHubLogWriter, IDisposable
+    {
+        private TraceSource _traceSource = null;
+
+        public DefaultLogHubLogWriter(TraceSource traceSource)
+        {
+            _traceSource = traceSource;
+        }
+
+        public override void Write(string message) =>
+            _traceSource?.TraceInformation(message);
```

Comment:
Why can trace source be `null`?

Reply:
Should not be, and should be asserted not null as part of the constructor. Thanks for calling it out!